lib/update/phase_coredns.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package update
+
+import (
+	"context"
+
+	"github.com/gravitational/gravity/lib/storage"
+	"github.com/gravitational/rigging"
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	corednsResourceName = "gravity:coredns"
+)
+
+func (r phaseBuilder) corednsPhase(leadMaster storage.Server) *phase {
+	phase := root(phase{
+		ID:          "coredns",
+		Description: "Provision coredns resources",
```

Comment:
Let's at least make it properly-cased (is that a word? :smile:) in the user visible string. "coredns" -> "CoreDNS"